File: 프로그래머스/lv2/49993. 스킬트리/스킬트리.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer =skill_trees.size();
    
    for (string skills:skill_trees) {
        vector<char> compare;
        for (char c:skills) {
            for (char s:skill) {
                if (c == s) {
                    compare.push_back(c);
                }
            }
        }
        for (int i=0; i<compare.size(); i++) {
            if (compare[i] != skill[i]) {
                answer--;
                break;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/lv2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = len(skill_trees)
    
    for skills in skill_trees:
        lst = []
        for c in skills:
            if c in skill:
                lst.append(skill.find(c))
        for i in range(len(lst)):
            if i != lst[i]:
                answer -= 1
                break
    return answer